zebra-network/src/peer_set/set/tests/vectors.rs

```diff
@@ -32,32 +31,23 @@ fn peer_set_drop() {
             .build();
 
         // Wait until the peer set is ready
-        let peer_set1 = peer_set.ready().now_or_never();
-
-        //
-        match peer_set1 {
-            Some(ready) => {
-                // make a call that returns a future
-                let fut = ready
-                    .expect("peer set service is always ready")
-                    .call(Request::Peers);
-                // drop the future
-                std::mem::drop(fut);
-
-                // Peer set will still be ready
-                let peer_set2 = peer_set.ready().now_or_never();
-                match peer_set2 {
-                    Some(ready2) => {
-                        // make a new call
-                        let _response = ready2
-                            .expect("peer set service is always ready")
-                            .call(Request::Peers)
-                            .now_or_never();
-                    }
-                    None => panic!("peer set service is always ready"),
-                }
-            }
-            None => panic!("peer set service is always ready"),
-        }
+        let peer_ready1 = peer_set
+            .ready()
+            .await
+            .expect("peer set service is always ready");
+
+        // make a call that returns a future
+        let fut = peer_ready1.call(Request::Peers);
+        // drop the future
+        std::mem::drop(fut);
+
+        // Peer set will still be ready
+        let peer_ready2 = peer_set
+            .ready()
+            .await
+            .expect("peer set service is always ready");
+
+        // make a new call
+        let _fut = peer_ready2.call(Request::Peers);
```

Comment:
Can we check that this future completes, rather than hanging?

Reply:
This future and the one in line 45 will both hang if we await on them.

Reply:
Can we return a fake response on the `Client` request channel using the test harness, and then check it is sent correctly via the `PeerSet`?

(If we return a fake response, it shouldn't hang.)

Reply:
Example of this can be found at https://github.com/ZcashFoundation/zebra/pull/3273 and we need https://github.com/ZcashFoundation/zebra/pull/3241 to be merged.

Reply:
I am marking this conversation as we are not going to make all the tests we initially wanted to do. We want to the focus in the last review where tests we already have are improved.